alchemist-web-renderer/src/commonMain/kotlin/it/unibo/alchemist/common/utility/Ruotes.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2010-2023, Danilo Pianini and contributors
+ * listed, for each module, in the respective subproject's build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.common.utility
+
+/**
+ * Objects that store all the route path. Since this object is part of the common sourceSet, both Client and Serve can
+ * refer to the correct Route maintaining consistency and reducing the number of possible errors and bug.
+ */
+object Ruotes {
+    /**
+     * Base environment path, for all Environment related operations.
+     */
+    private const val environmentPath: String = "/environment"
```

Comment:
Question: how do we deal with multiple simulations (hence, multiple environments) running on the server?

Reply:
**_alchemist-web-renderer_** takes one configuration file and run the simulation while also offering a web server for accessing its state. The easiest way that comes to my mind is just to launch _n_ instances of _web-renderer_ on the server. Every instance would take a different port and work as expected. In this case the limitation would be on the front-end, as two different tab are needed on the browser.

Reply:
What happens if I launch a batch?
Alchemist configuration files do not describe a single simulation, but a batchable set of them

Reply:
Batch mode is still not supported. It would still be possible to implement it, but the ServerState and the Routes Logic should be modified a bit . Finally a Button to switch between the Simulations on the front-end should be added.
With the current architecture this change should not be that hard to achieve, but it is still not present.
I tried inspiration from the Swing Gui and the Fx gui, since this project is similar from certain points, and as I see (correct me if I'm wrong) none of them supports the batch mode.
Anyway I'm keeping tracks of all the features that are still missing from the beginning of the project and how they could be achieved by changing the existent codebase. If you think that's okay I would add them to the "Future work" section of my thesis, so anyone who would work on this project after me would have a easier life.

Reply:
@AngeloFilaseta ok. The difference with the other two UIs is that this one is distributed, and the server is meant to run on a capable machine, with users on the network that may want to run simulations on it concurrently. Add this feature to the top of the future work, as soon as we have the current state merged and the thesis mostly written, if there is time I'd like to tackle this issue.